src/utils/operation.ts

```diff
@@ -142,14 +142,15 @@ export function sendTransaction<T>(
     } else {
       try {
         gasLimit = (await contract.estimate[tx.method](...tx.args, tx.opts)).toNumber()
+        gasLimit *= 2
       } catch (error) {
         await catchHandler(error, tx, await signer.getAddress())
       }
     }
 
     const overrides = {
       ...tx.opts,
-      gasLimit: gasLimit ? gasLimit : 1000000
+      gasLimit: gasLimit ? Math.min(1000000, gasLimit) : 1000000
```

Comment:
so the assumption is that the gasLimit for any dao operation will not exceed 1000000 ?
If there is a plan (or it is currently the case) that daoFactory will be used via arc.js so this might not be enough...for registering >x  schemes or forging a dao with many members.

Reply:
seems to me this should be more like the block size limit instead of 1M.

But in any case, whatever the number is, this logic will fail if the gas usage is > 1M, as oren says.

Reply:
Yep that makes sense, this "min(...)" logic came from the original here: https://github.com/daostack/arc.js/blob/05b3b176b00c0ea3a156e701a6ae5577c0f8c62d/src/operation.ts#L112